tkg/managementcomponents/management_component_install.go

```diff
@@ -290,14 +291,18 @@ func InstallManagementComponents(mcip *ManagementComponentsInstallOptions) error
 
 // InstallKappController installs kapp-controller to the cluster
 func InstallKappController(clusterClient clusterclient.Client, kappControllerOptions KappControllerOptions) error {
+	// Ensure last-applied annotation is present before kapp-controller upgrade
+	if err := clusterClient.GetResource(&appsv1.Deployment{}, constants.KappControllerDeploymentName, kappControllerOptions.KappControllerInstallNamespace, nil, nil); err == nil { // only attempt to add annotation when kapp-controller exists on the cluster
```

Comment:
**Thoughts**: Thinking about this I think it might be better to throw error if it is not NotFound error.

e.g. 
```
err := clusterClient.GetResource(&appsv1.Deployment{}, constants.KappControllerDeploymentName, kappControllerOptions.KappControllerInstallNamespace, nil, nil)
if err != nil && !apierrors.IsNotFound(err) {
   return err
 }
 if err == nil {
   if err := clusterClient.PatchKappControllerLastAppliedAnnotation(kappControllerOptions.KappControllerInstallNamespace); err != nil {
			return errors.Wrap(err, "error adding last-applied annotation on kapp-controller")
		}
}
// Apply kapp-controller configuration
	if err := clusterClient.ApplyFile(kappControllerOptions.KappControllerConfigFile); err != nil
```